amdahl/amdahl.py

```diff
@@ -110,19 +119,24 @@ def parse_command_line():
 def amdahl():
     """Amdahl's law illustrator (with fake work)"""
     rank = MPI.COMM_WORLD.Get_rank()
-    # Only the root process handles the command line arguments
+    # All processes handle (at least some) command line arguments
+    args = parse_command_line()
     if rank == 0:
         # Start a clock to measure total time
         start = time.time()
 
-        args = parse_command_line()
-
         do_work(
-            work_time=args.work_seconds, parallel_proportion=args.parallel_proportion
+            work_time=args.work_seconds, parallel_proportion=args.parallel_proportion,
+            terse=args.terse
         )
         end = time.time()
-        sys.stdout.write(
-            "\nTotal execution time (according to rank 0): %f seconds\n" % (end - start)
-        )
+        if args.terse:
+            sys.stdout.write(
+                "%f\n" % (end - start)
```

Comment:
This is a little bit too terse for my taste as there's no context at all for this number. For me the terse option should just silence all non-root MPI processes. That means you get a guaranteed output sequence that you can parse for your data.

A bonus would be to make that output really easy to parse (e.g. make it json):
```
{
  'serial_work':  '',
  'parallel_work': ''.
  'parallel_proportion': '',
  'execution_time' '',
}
```

Reply:
Should be able to make the JSON work.

As for root versus non-root MPI ranks printing things, do you really want the root rank printing:

- "Doing %f seconds of 'work' on %s processor %s, which should take %f seconds with %f parallel proportion of the workload."
- "Hello, World! I am process %d of %d on %s. I will do all the serial 'work' for %f seconds."
- "Hello, World! I am process %d of %d on %s. I will do parallel 'work' for %f seconds."

or do you want it to just print the summary data as shown in the JSON snippet?

Reply:
If you're doing the JSON then that's all that's required